### PR TITLE
fix(rome_js_analyze): better diagnostic message of useSimplifiedLogicExpression

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
## Summary
1. Give a better message based on the case

## Test Plan
1. cargo test

### CODE REVIEW

**Comment by ematipico** [comment]
@IWANABETHATGUY can you try to update your upstream branch? It seems your fork still has old code, causing the the PR to fail

File: crates/rome_js_analyze/src/analyzers/use_simplified_logic_expression.rs [hunk 1]
```diff
@@ -56,7 +56,8 @@ impl Rule for UseSimplifiedLogicExpression {
     const CATEGORY: RuleCategory = RuleCategory::Lint;
 
     type Query = Ast<JsLogicalExpression>;
-    type State = JsAnyExpression;
+    /// First element of tuple is if the expression is simplified by `de_morgan` rule, second is expression after replaced.
```
Comment (ematipico) [inline crates/rome_js_analyze/src/analyzers/use_simplified_logic_expression.rs:59]:
```suggestion
    /// First element of tuple is if the expression is simplified by [De Morgan's Law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) rule, the second element is the expression to replace.
```

File: crates/rome_js_analyze/src/analyzers/use_simplified_logic_expression.rs [hunk 2]
```diff
@@ -130,14 +133,20 @@ impl Rule for UseSimplifiedLogicExpression {
...
     fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
         let node = ctx.query();
+        let (is_simplified_by_de_morgan, expr) = state;
         let root = ctx.root().replace_node(
             JsAnyExpression::JsLogicalExpression(node.clone()),
-            state.clone(),
+            expr.clone(),
         )?;
+        let message = if *is_simplified_by_de_morgan {
+            "Reduce the complexity of LogicExpression."
```
Comment (ematipico) [inline crates/rome_js_analyze/src/analyzers/use_simplified_logic_expression.rs:142]:
```suggestion
            "Reduce the complexity of the logical expression."
```

File: crates/rome_js_analyze/src/analyzers/use_simplified_logic_expression.rs [hunk 3]
```diff
@@ -130,14 +133,20 @@ impl Rule for UseSimplifiedLogicExpression {
...
+        } else {
+            "Discard redundant terms from logical expressions"
```
Comment (ematipico) [inline crates/rome_js_analyze/src/analyzers/use_simplified_logic_expression.rs:144]:
```suggestion
            "Discard redundant terms from the logical expression."
```

**Review by ematipico — CHANGES_REQUESTED** [review]

**Comment by IWANABETHATGUY** [comment]
> @IWANABETHATGUY can you try to update your upstream branch? It seems your fork still has old code, causing the the PR to fail

I have rebased main branch.

**Comment by IWANABETHATGUY** [comment]
I have no idea to pass the CI, feel free to change my code.

**Pull request merged**